sympy/ntheory/residue_ntheory.py

```diff
@@ -1498,7 +1499,10 @@ def _valid_expr(expr):
     polynomial = Poly(expr)
     if not  polynomial.is_univariate:
         raise ValueError("The expression should be univariate")
-    return polynomial.all_coeffs()
+    coefficients = polynomial.all_coeffs()
```

Comment:
It should suffice to check  `polynomial.domain == ZZ`.

Reply:
Done.